### ISSUE TITLE
`PublicMemberInNonPublicType` does not detect overridden methods

### ISSUE THREAD

**Issue by uhafner** [pmd/pmd#6460]
**Affects PMD Version: 7.22** 

**Rule: [PublicMemberInNonPublicType](https://pmd.github.io/pmd/pmd_rules_java_design.html#publicmemberinnonpublictype)**

**Description:**

The new rule does not flag methods, when they are overridden in sub classes with the wrong visibility. 

**Code Sample demonstrating the issue:**

```java
class Outer {
    public void detected() {
        // this method is correctly detected
    }

    void notDetected() {
        // this method is correctly not detected here because it is package private
    }

    static class Nested extends Outer {
        public void nestedMethod() {
            // this method is correctly detected
        }

        @Override
        public void notDetected() {
            // this one should be detected now, but is not
        }
    }
}
```

**Expected outcome:**

PMD should report 3 violations, but does report only 2. 

```
[WARNING] PMD Failure: edu.hm.hafner.grading.Outer:4 Rule:PublicMemberInNonPublicType Priority:3 Public member 'detected' declared in a non-public type.
[WARNING] PMD Failure: edu.hm.hafner.grading.Outer.Nested:13 Rule:PublicMemberInNonPublicType Priority:3 Public member 'nestedMethod' declared in a non-public type.
```

**Running PMD through:** *Maven*
